src/CMakeLists.txt

```diff
@@ -9,49 +9,51 @@ add_library(musica::micm ALIAS micm)
 
 target_compile_features(micm INTERFACE cxx_std_20)
 
+target_compile_definitions(micm INTERFACE MICM_DEFAULT_VECTOR_SIZE=${MICM_DEFAULT_VECTOR_MATRIX_SIZE})
+
 target_include_directories(micm
   INTERFACE
     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
     $<INSTALL_INTERFACE:include>
 )
 
-if(ENABLE_JSON)
+if(MICM_ENABLE_JSON)
   target_link_libraries(micm INTERFACE nlohmann_json::nlohmann_json)
   target_compile_definitions(micm INTERFACE USE_JSON)
 endif()
 
-if(ENABLE_OPENMP)
+if(MICM_ENABLE_OPENMP)
   target_link_libraries(micm INTERFACE OpenMP::OpenMP_CXX)
 endif()
 
-if(ENABLE_MPI)
+if(MICM_ENABLE_MPI)
   target_link_libraries(micm INTERFACE MPI::MPI_CXX)
 endif()
 
-if(ENABLE_LLVM)
+if(MICM_ENABLE_LLVM)
   target_link_libraries(micm INTERFACE ${llvm_libs})
 endif()
 
-if(ENABLE_OPENACC OR ENABLE_CUDA)
-  if(NOT GPU_TYPE)
-    message(FATAL_ERROR "GPU_TYPE is not set or is empty. Please provide a GPU type.")
+if(MICM_ENABLE_OPENACC OR MICM_ENABLE_CUDA)
+  if(NOT MICM_GPU_TYPE)
+    message(FATAL_ERROR "MICM_GPU_TYPE is not set or is empty. Please provide a GPU type.")
   endif()
 
   set(GPU_ARCH "")
-  if (GPU_TYPE STREQUAL a100)
+  if (MICM_GPU_TYPE STREQUAL a100)
     set(GPU_ARCH "80")
```

Comment:
We should also set `GPU_ARCH` to `MICM_GPU_ARCH` and then I can re-arrange how it's used when I clean up the cuda project cmake to use the newer `project(...)` syntax.

Reply:
If this would break the current scripts, I would suggest holding off until you submit the PR with the cleaned-up CUDA build

Reply:
@mwaxmonsky done